R/human_infection.R

```diff
@@ -183,21 +184,96 @@ calculate_infections <- function(
   }
 
   prob <- b * (1 - prophylaxis) * (1 - vaccine_efficacy)
-  infected <- bitset_at(source_humans, bernoulli_multi_p(prob))
 
-  incidence_renderer(
-    variables$birth,
-    renderer,
-    infected,
-    source_humans,
-    prob,
-    'inc_',
-    parameters$incidence_rendering_min_ages,
-    parameters$incidence_rendering_max_ages,
-    timestep
-  )
+  ## All bitten humans with an infectious bite (incorporating prophylaxis and vaccines)
+  newly_infected <- bitset_at(source_humans, bernoulli_multi_p(prob))
+
+  if(parameters$parasite == "falciparum"){
+    # Render new infections caused by bites
+    renderer$render('n_infections', newly_infected$size(), timestep)
+    incidence_renderer(
+      variables$birth,
+      renderer,
+      newly_infected,
+      source_humans,
+      prob,
+      'inc_',
+      parameters$incidence_rendering_min_ages,
+      parameters$incidence_rendering_max_ages,
+      timestep
+    )
+    total_infected <- newly_infected
+  }
 
-  infected
+  else if(parameters$parasite == "vivax"){
+
+    SAU_humans <- variables$state$get_index_of(c('S','A','U'))
+    new_bite_infections <- newly_infected$and(SAU_humans)
+
+    # Render new infections caused by bites
+    renderer$render('n_new_bite_infections', new_bite_infections$size(), timestep)
+    incidence_renderer(
+      variables$birth,
+      renderer,
+      new_bite_infections,
+      source_humans,
+      prob,
+      'inc_new_bite_',
+      parameters$new_bite_incidence_rendering_min_ages,
+      parameters$new_bite_incidence_rendering_max_ages,
+      timestep
+    )
+
+    ## All bitten humans with an infectious bite (incorporating prophylaxis and vaccines get a new batch of hypnozoites)
+    variables$hypnozoites$queue_update(
+      variables$hypnozoites$get_values(newly_infected) + 1,
+      newly_infected
+    )
```

Comment:
We have a competing hazards issues here.
In Michaels model, from your description, they draw all infections, and then split those into either bites or relapses.
In this implementation you draw infections from bites first, and all of those get +1 hypnozoite batches. This will be different from Michaels implementation because a proportion of those new infections from bites would actually be relapses and not get +1 hypnozoite batches. 
Given how sensitive the steady state seems to be regarding hypnozoite dynamics, this could be influential and is therefore something we probably want to align on. In general that would be drawing individuals who get either new bite infection or relapse infection, then splitting those between new bites or relapses based on the relative rates of each.

Reply:
I've had a go at this. Let me know if there's anything that I could improve.

Reply:
^Had another go - I think I've corrected the previous errors now.